src/hotspot/os_cpu/bsd_x86/os_bsd_x86.cpp

```diff
@@ -401,12 +401,8 @@ JVM_handle_bsd_signal(int sig,
 
   Thread* t = Thread::current_or_null_safe();
 
-  // Must do this before SignalHandlerMark, if crash protection installed we will longjmp away
```

Comment:
Should we keep a comment (and add to other files) to the effect that:

// If crash protection is installed we will longjmp away and no destructors from any RAII utility
// object would be run. So don't use any RAII utilities here.

?

Reply:
```
  // If crash protection is installed we may longjmp away and no destructors
  // for objects in this scope will be run.
  // So don't use any RAII utilities before crash protection is checked.

```
Since even if crash protection is installed it's not certain we will longjmp and you can trust destructors after we checked crash protection. ?

Reply:
I'm not sure if only the call to `check_crash_protection` can trigger the long jump? If so then your version of the comment is better.

Reply:
Yes only check_crash_protection() do the longjmp and only if it's the protected thread and on SIGSEGV/SIGBUS.